File: 1363-greatest-english-letter-in-upper-and-lower-case/greatest-english-letter-in-upper-and-lower-case.java
class Solution {
    public String greatestLetter(String s) {
        String ans = "";
        for(char ch : s.toCharArray()){
            String str = Character.toUpperCase(ch)+"";
            if(Character.isUpperCase(ch)){
                String chh = Character.toLowerCase(ch) +"";
                if(s.contains(chh)){
                    if(str.compareTo(ans) > 0){
                        ans = str;
                    }
                }
            }
        }
        return ans;
    }
}

File: 1363-greatest-english-letter-in-upper-and-lower-case/greatest-english-letter-in-upper-and-lower-case.py
class Solution:
    def greatestLetter(self, s: str) -> str:
        l = set(s)  # no need for list() before set
        ans = ''
        for i in s:
            if 'a' <= i <= 'z':  
                if i.upper() in l:
                    ans = max(ans, i.upper())
            elif 'A' <= i <= 'Z':  
                if i.lower() in l:
                    ans = max(ans, i)
        return ans
